### DISCUSSION TITLE
Implications of ag-solo vs. smart wallet architecture on Agoric smart contracts

### DISCUSSION THREAD

**Discussion by migueldingli1997** [Agoric/agoric-sdk#8584] [Q&A]
As a newcomer to smart contract development on Agoric, I would like to understand more concretely the implications of the [transition from ag-solo to the smart wallet architecture](https://github.com/Agoric/documentation/issues/726), especially on how smart contracts are developed/structured/deployed/etc.

I've been going through the [getting started docs](https://docs.agoric.com/guides/getting-started). Knowing about the above made me question some of the documentation such as the [Agoric Wallet page](https://docs.agoric.com/guides/wallet/) since it might get phased out eventually due to the mention of ag-solo.

The [ag-solo based docs](https://docs.agoric.com/guides/getting-started/start-a-project.html) points out that:

> The Dapp architecture presented here is a beta preview of our eventual permissionless contract deployment model, extending our Distributed Computing Framework to a stateful peer on end-user machines.
>
> The Mainnet-1 launch uses a "smart wallet" architecture with a lower client-side footprint.

I was also made aware that ag-solo style deployment is supported in devnet but not in mainnet. This makes me wonder about how I should be approaching smart contract development given that I want to be developing contracts that run on both the devnet and mainnet. It also makes me wonder about what I should be on the lookout for if I'm looking at example contracts, given that these might have been developed with ag-solo in mind.

**Comment by dckc** [comment]
One-shot contracts like the [PSM](https://docs.agoric.com/guides/zoe/actual-contracts/PSM.html) work well.

For longer flows like [vaults](https://community.agoric.com/t/inter-protocol-vaults-contract-implementations/261/2), the continuing invitation pattern is well supported... though documentation is scattered:

  - https://github.com/Agoric/documentation/issues/667

One aspect of several of our example contracts that's not supported by the smart wallet architecture is invitations as offer results; here's hoping we remove that limitation before long:

 - https://github.com/Agoric/agoric-sdk/issues/8550
 
 _Here's hoping for time to think of a more exhaustive answer._
